src/Cake.Tfs/TfsAliases.PullRequest.cs

```diff
@@ -55,6 +55,34 @@ public static partial class TfsAliases
             return null;
         }
 
+        /// <summary>
+        /// Gets a Team Foundation Server or Azure DevOps pull request using the settings obtained from predefined build variables.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <example>
+        /// <para>Get a pull request:</para>
+        /// <code>
+        /// <![CDATA[
+        ///     var pullRequest = TfsPullRequestUsingTfsBuildOAuthToken();
+        /// ]]>
+        /// </code>
+        /// </example>
+        /// <returns>Description of the pull request.
+        /// Returns null if pull request could not be found and
+        /// <see cref="TfsPullRequestSettings.ThrowExceptionIfPullRequestCouldNotBeFound"/> is set to <c>false</c>.</returns>
+        /// <exception cref="TfsPullRequestNotFoundException">If pull request could not be found and
+        /// <see cref="TfsPullRequestSettings.ThrowExceptionIfPullRequestCouldNotBeFound"/> is set to <c>true</c>.</exception>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Pull Request")]
+        public static TfsPullRequest TfsPullRequestUsingTfsBuildOAuthToken(
```

Comment:
As this doesn't make sense on on-prem TFS TfsPullRequestUsingAzurePipelinesOAuthToken might be a more appropriate name

Reply:
Hmm, as far as I know, it is possible to use access token in on-prem TFS too. At least, if I go to the build definition, select Phase on the left pane, then Additional Options at the bottom of the right pane contain the checkbox "Allow Scripts to Access OAuth Token". It's TFS 2018.2. Am I missing anything?

![1](https://user-images.githubusercontent.com/3975939/46274359-d7556e80-c561-11e8-9a51-4cc84a87208c.png)

Reply:
Interesting. I thought the whole OAuth workflow doesn't exist on-premise. Would need to test this out, and if OAuth credentials need to be used on-prem or if it needs to be passed as personal access token.